clippy_lints/src/trait_bounds.rs

```diff
@@ -0,0 +1,73 @@
+use crate::utils::{in_macro, snippet, span_help_and_lint, SpanlessHash};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, impl_lint_pass};
+use rustc_data_structures::fx::FxHashMap;
+use rustc::hir::*;
+
+#[derive(Copy, Clone)]
+pub struct TraitBounds;
+
+declare_clippy_lint! {
+    /// **What it does:** This lint warns about unnecessary type repetitions in trait bounds
+    ///
+    /// **Why is this bad?** Complexity
```

Comment:
This should be a little bit more explanatory. 
```suggestion
    /// **Why is this bad?** Repeating the type for every bound makes the code less readable than combining the bounds.
```

Or something similar :)